modules/data.land/R/InventoryGrowthFusion.R

```diff
@@ -224,37 +265,91 @@ model{
   if(!is.null(time_varying)){
     if (is.null(time_data)) {
-      print("time_varying formula provided but time_data is absent:", time_varying)
+      PEcAn.utils::logger.error("time_varying formula provided but time_data is absent:", time_varying)
     }
+    Xt.priors <- ""
     
     ## parse equation into variable names
     t_vars <- gsub(" ","",unlist(strsplit(time_varying,"+",fixed=TRUE))) ## split on +, remove whitespace
     ## check for interaction terms
-    it_vars <- grep(pattern = "*",x=t_vars,fixed = TRUE)
-      ## need to deal with interactions with fixed variables
-      ## will get really nasty if interactions are with catagorical variables
-      ## need to create new data matrices on the fly
+    it_vars <- t_vars[grep(pattern = "*",x=t_vars,fixed = TRUE)]
+    if(length(it_vars) > 0){
+      t_vars <- t_vars[!(t_vars %in% it_vars)]
+    } 
+    
+    ## INTERACTIONS WITH TIME-VARYING VARS
+    ## TODO: deal with interactions with catagorical variables
+    ## need to create new data matrices on the fly
+    for(i in seq_along(it_vars)){
+
+      ##is covariate fixed or time varying?
+      covX <- strsplit(it_vars[i],"*",fixed=TRUE)[[1]] 
+      tvar    <- length(grep("[t]",covX[1],fixed=TRUE)) > 0
+      tvar[2] <- length(grep("[t]",covX[2],fixed=TRUE)) > 0
+      myBeta <- "beta"
+      for(j in 1:2){
+        if(j == 2) myBeta <- paste0(myBeta,"_")
+        if(tvar[j]){
+          covX[j] <- sub("[t]","",covX[j],fixed = TRUE)
+          if(!(covX[j] %in% names(data))){
+            ## add cov variables to data object
+            data[[covX[j]]] <- time_varying[[covX[j]]]
+          }
+          myBeta <- paste0(myBeta,covX[j])
+          covX[j] <- paste0(covX[j],"[i,t]")
+        } else {
+          ## variable is fixed
+          if(!(covX[j] %in% names(data))){
+            ## add cov variables to data object
+            data[[covX[j]]] <- cov.data[,covX[j]]
+          }
+          myBeta <- paste0(myBeta,covX[j])
+          covX[j] <- paste0(covX[j],"[i]")
+        } ## end fixed or time varying
+        
+      } ## end building beta
+      
+      ## append to process model formula
+      Pformula <- paste(Pformula,
+                        paste0(" + ",myBeta,"*",covX[1],"*",covX[2]))
+      
+      ## priors
+      Xt.priors <- paste0(Xt.priors,
+                          "    ",myBeta,"~dnorm(0,0.001)\n")
+      
+      ## add to list of varibles JAGS is tracking
+      out.variables <- c(out.variables, myBeta)
+      
+    }  ## end time-varying interaction terms
+    
     
     ## loop over variables
     for(j in seq_along(t_vars)){
       tvar <- t_vars[j]
       
-      ## grab from the list of data matrices
-      dtmp <- time_data[[tvar]]
+      if(!(tvar %in% names(data))){
+        ## add cov variables to data object
+        data[[tvar]] <- time_data[[tvar]]
+      }
+      if(any(duplicated(names(data)))){PEcAn.utils::logger.error("duplicated variable at tvar",names(data))}
```

Comment:
This block has happened above too, so it might be worth refactoring into a simple function? I mean the whole block, not just the error statement.

Also, should this be `logger.severe`, since `logger.error` doesn't actually throw an R error?